### PR TITLE
Moving to typescript pt 1

### PR DESCRIPTION
**Pull request by mariabug**

### CODE REVIEW

**Comment by mariabug** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #20** <a href="https://app.graphite.dev/github/pr/DonArtisan/technify-react/20" target="_blank">Graphite</a> 
    * **PR #21** <a href="https://app.graphite.dev/github/pr/DonArtisan/technify-react/21" target="_blank">Graphite</a> 
      * **PR #22** <a href="https://app.graphite.dev/github/pr/DonArtisan/technify-react/22" target="_blank">Graphite</a>  👈
        * **PR #23** <a href="https://app.graphite.dev/github/pr/DonArtisan/technify-react/23" target="_blank">Graphite</a> 
          * **PR #24** <a href="https://app.graphite.dev/github/pr/DonArtisan/technify-react/24" target="_blank">Graphite</a> 
            * **PR #25** <a href="https://app.graphite.dev/github/pr/DonArtisan/technify-react/25" target="_blank">Graphite</a> 
              * **PR #26** <a href="https://app.graphite.dev/github/pr/DonArtisan/technify-react/26" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DonArtisan/technify-react/22?utm_source=stack-comment).

File: src/components/ProductCard.tsx [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import {Button, Flex, Image, Stack, Text} from '@chakra-ui/react'
 import {useCartActions} from '../stores/useCartStore'
-import defalutImg from '../utils/assets/image/default-vs.jpg'
+import {Product} from '../../generated/graphql'
 
-export default function ProductCard({product}) {
+interface ProductCardProps {
+  product: Product
```
Comment (AguilarKevin) [inline src/components/ProductCard.tsx]: podes usar el helper `Pick` para que evites hacer la asercion `product as Product`

File: src/pages/home/index.tsx [hunk 2]
```diff
@@ -13,12 +13,14 @@ import {FiHeadphones} from 'react-icons/fi'
 import {IoIosPricetag, IoMdContact} from 'react-icons/io'
 import {graphql, useLazyLoadQuery} from 'react-relay'
 import {Link as ReactRouterLink} from 'react-router-dom'
+import {Product} from '../../../generated/graphql'
 import CategoriesList from '../../components/CategoriesList'
```
Comment (AguilarKevin) [inline src/pages/home/index.tsx]: podes agregar un path alias para evitar poner todo el `../../../`

File: src/pages/home/index.tsx [hunk 3]
```diff
@@ -87,11 +90,13 @@ export default function Home() {
         </Flex>
         <Flex width="full" paddingBlock="20px" gap="32px" overflowX="scroll">
           {products?.edges.map(({node: product}, index) => (
-            <ProductCard key={index} product={product} />
+            <ProductCard key={index} product={product as Product} />
```
Comment (AguilarKevin) [inline src/pages/home/index.tsx]: usar el index del item en el array es un anti-pattern porque el `key` prop debe ser unicom, podrias usar el id del `product`

**Review by AguilarKevin — CHANGES_REQUESTED** [review]

**Pull request merged**